packages/core/src/components/Editable/index.js

```diff
@@ -79,10 +81,12 @@ class Editable extends Component {
     const state: any = editable(this.props.editor.store.getState(), {
       id: this.props.id
     })
-    if (state === this.previousState || !state) {
+
+    if (isEqual(state, this.previousState) || !state) {
```

Comment:
This may have some serious performance issues, as this compares every key in the tree with one another. In large documents, this will cause a ton of comparisons. If you type fast, it will cause lags on slow systems. A better idea would be to have immutable objects and do a simple comparison instead.

Reply:
What about using JSON.stringify ? Ill try to do some tests

Reply:
It will be even worse. What we really should use is [immutable.js](https://facebook.github.io/immutable-js/). You can however use this fix in your own code to prevent the issue you're experiencing. I don't think it makes sense to merge this patch here though.